pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/validator/ModelValidatorImpl.groovy

```diff
@@ -343,16 +342,21 @@ class ModelValidatorImpl implements ModelValidator {
                     List<DescribableParameter> requiredParams = model.parameters.findAll { it.isRequired() }
 
                     if (requiredParams.size() != meth.args.size()) {
-                        errorCollector.error(meth, Messages.ModelValidatorImpl_WrongNumberOfStepParameters(meth.name, requiredParams.size(), meth.args.size()))
-                        valid = false
+                        // TODO: This is a specialized hack for allowing single-required-Boolean-parameter constructors
```

Comment:
What is the TODO saying we'll need to do eventually?  Whatever it is file a JIRA and reference it.  Otherwise, it is just a comment not a TODO.